src/main/java/codeu/model/store/basic/IndexStore.java

```diff
@@ -66,13 +71,88 @@ public void addMapping (String word, Message message) {
   /**
     * Search function for a single word, returns a Set of messages from the Hashtable if the word is in it, otherwise return null
     */
-  public Set<Message> searchWord (String word) {
-    if (wordsHash.containsKey(word)) {
-      return wordsHash.get(word);
+  public List<Message> search (String searchTarget) {
+    if (searchTarget.contains("||")) {
+      return searchUnion(searchTarget);
+    }else if (searchTarget.contains("&&")) {
+      searchTarget = searchTarget.replaceAll(" ","");
```

Comment:
I think this woudl be nice if it also replaced whitespace with &&s.  Currently this turns "hello && world foo" into "hello && worldfoo"  I think implied conversion into "hello && world && foo" makes more sense.